examples/cluster.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python3
+
+from urllib.request import Request, urlopen, HTTPError
+import json
+
+
+def request(method, url, data=None, headers=None):
+    """Make a request against the JSON url
+    :param method: the request method
+    :param url: the path in the JSON API
+    :param data: the request body
+    :type data: bytes|str
+    """
+
+    headers = dict(headers or {})
+    body = None
+
+    if data is not None:
+        if isinstance(data, bytes):
+            body = data
+        else:
+            headers['Content-Type'] = 'application/json'
+            body = json.dumps(data).encode('utf8')
+
+    req = Request(
+        url,
+        body,
+        headers,
+        method=method,
+    )
+
+    with urlopen(req) as res:
+        if res.status >= 200 and res.status < 300:
+            content = res.read()
+            if (res.headers.get('Content-Type') == 'application/json' and
+                    len(content)):
+                return json.loads(content)
+            else:
+                return None
+        else:
+            raise HTTPError(url, res.status, res.msg, res.headers, res)
+
+
+def main():
+    job_vmck = {}
+    data = {}
+
+    # nomad hcl->json convertor
+    url_parser = 'http://10.66.60.1:4646/v1/jobs/parse'
+    # nomad api for job submission
+    url_submit = 'http://10.66.60.1:4646/v1/jobs'
+
+    data['Canonicalize'] = True
+    data['JobHCL'] = open('vmck.nomad', 'r').read()
```

Comment:
It's impolite to do `open(...).read()` because it leaves closing the file as a side effect of the memory manager. And `'r'` is the default. Better to write:
```py
with open('vmck.nomad') as f:
    job_hcl = f.read()
```